geode-core/src/main/java/org/apache/geode/cache/configuration/RegionConfig.java

```diff
@@ -167,6 +167,13 @@
   @XmlAttribute(name = "refid")
   protected String refid;
 
+  public RegionConfig() {}
+
+  public RegionConfig(String name, String refid) {
```

Comment:
This class is not special enough to get a parameterized constructor that no other config object gets.

Reply:
It's just a convenience constructor.

Reply:
I made a similar comment on another PR, but it's the inconsistency that bothers me.  If we want our configuration objects to be declarative, we should have them be declarative.  If we want convenient, parameterized constructors, we should go through and add them all rather than selectively doing it piecemeal.

Reply:
We don't have to add them all at once though. We can always add them when needed. "name" and "refid" are the only requirement attributes of a region, hence this constructor is useful.